### PR TITLE
Move common verify_deployed_contract() usage into the function body

### PR DESCRIPTION
**Pull request by pirapira**
the function body.

This is one proposal for solving https://github.com/raiden-network/raiden-contracts/issues/502.

I'm not super-confident with the direction I took: contract-specific checks as function arguments.

### CODE REVIEW

File: raiden_contracts/deploy/__main__.py [hunk 1]
```diff
@@ -744,99 +729,70 @@ def verify_deployed_service_contracts(
...
-    assert token_address == constructor_arguments[0]
-
-    print(
-        f'{CONTRACT_SERVICE_REGISTRY} at {service_bundle.address} '
-        f'matches the compiled data from contracts.json',
-    )
-
+    def user_deposit_checker(deploy_record, onchain_instance):
+        # We need to also check the constructor parameters against the chain
+        constructor_arguments = deploy_record['constructor_arguments']
```
Comment (karlb) [inline raiden_contracts/deploy/__main__.py]: This line is in all checkers, so `verify_deployed_contract` could call each checker directly with `constructor_arguments`.

File: raiden_contracts/deploy/__main__.py [hunk 2]
```diff
@@ -664,55 +664,40 @@ def verify_deployed_contracts(web3: Web3, contract_manager: ContractManager, dep
...
+        ) == secret_registry.address
+        assert secret_registry.address == constructor_arguments[0]
+
+        chain_id = onchain_instance.functions.chain_id().call()
+        assert chain_id == constructor_arguments[1]
+
+        settlement_timeout_min = onchain_instance.functions.settlement_timeout_min().call()
+        settlement_timeout_max = onchain_instance.functions.settlement_timeout_max().call()
+        assert settlement_timeout_min == constructor_arguments[2]
+        assert settlement_timeout_max == constructor_arguments[3]
```
Comment (karlb) [inline raiden_contracts/deploy/__main__.py]: I would do the assert directly if it fits in a single line, but that's a matter of taste:
```python
assert onchain_instance.functions.settlement_timeout_max().call() == constructor_arguments[3]
```

**Comment by pirapira** [comment]
@karlb 

> but if you want to avoid passing checker functions around, you could do

I like the return-value approach.  So, I will try that and open the PR again.

File: raiden_contracts/deploy/__main__.py [hunk 3]
```diff
@@ -744,99 +729,70 @@ def verify_deployed_service_contracts(
...
+        assert to_checksum_address(
+            onchain_instance.functions.deposit_contract().call(),
+        ) == user_deposit.address
+        assert user_deposit.address == constructor_arguments[0]
+        assert len(constructor_arguments) == 1
+    verify_deployed_contract(
         web3,
         contract_manager,
         deployment_data,
         CONTRACT_ONE_TO_N,
```
[the remark this answers is not in the document]
Comment (pirapira) [reply raiden_contracts/deploy/__main__.py]: I take this as an indication that I took a too fancy approach.

File: raiden_contracts/deploy/__main__.py (hunk 1 above)
Comment (pirapira) [reply raiden_contracts/deploy/__main__.py]: Yes.

I still wonder why a special treatment is necessary for constructor arguments, but sure, that's the way it is.

**Pull request closed without merging**